testing/test_many.sh

```diff
@@ -3,7 +3,8 @@
 source testing/test_preamble.sh
 
 NUM_DEVICES=8
-
+RUN_LIMIT=10
+NO_DHCP_DEVICES=1
```

Comment:
NUM_NO_DHCP_DEVICES -- otherwise I'd interpret this variable as a boolean, but you're using it in a comparison.